### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by derekdreery**
Just adds some docs matching what @matklad told me on irc :)

### CODE REVIEW

**Comment by derekdreery** [comment]
r? matklad

File: src/cargo/core/package_id_spec.rs [hunk 1]
```diff
@@ -16,6 +25,27 @@ pub struct PackageIdSpec {
...
+    ///     "crates.io/foo",
+    ///     "crates.io/foo#1.2.3",
+    ///     "crates.io/foo#bar",
+    ///     "crates.io/foo#bar:1.2.3",
+    ///     "foo",
+    ///     "foo:1.2.3",
+    /// ];
+    /// for spec in specs {
+    ///     assert!(PackageIdSpec::parse(spec).is_ok());
+    /// }
```
Comment (matklad) [inline src/cargo/core/package_id_spec.rs]: :heart_eyes:

File: src/cargo/core/package_id_spec.rs [hunk 2]
```diff
@@ -107,20 +141,25 @@ impl PackageIdSpec {
...
+    /// Getter for `version` field.
     pub fn version(&self) -> Option<&Version> {
         self.version.as_ref()
     }
+    /// Getter for `url` field.
     pub fn url(&self) -> Option<&Url> {
         self.url.as_ref()
     }
 
+    /// Setter for `url` field.
```
Comment (matklad) [inline src/cargo/core/package_id_spec.rs]: Hm, let's remove such trivial docs perhaps?

Comment (matklad) [reply src/cargo/core/package_id_spec.rs]: I mean, for getters and setters specifically, everything else looks great!

**Comment by matklad** [comment]
Thanks!

@bors r+

**Pull request merged**